packages/ui/src/components/cart/CartPopoverRenderer.tsx

```diff
@@ -87,37 +102,41 @@ export const CartPopoverRenderer: FC<Props> = ({ open, children }) => {
         ({ previousPrice, price }) =>
           previousPrice &&
           price?.amount?.decimal !== undefined &&
-          previousPrice.amount?.decimal !== price?.amount?.decimal
+          previousPrice.amount?.decimal !== price?.amount?.decimal &&
+          previousPrice.currency?.contract === price?.currency?.contract
       ),
     [items]
   )
   const { address } = useAccount()
-  const { data: balance } = useBalance({
-    chainId: cartChain?.id || client?.currentChain()?.id,
-    address: address,
-    token:
-      currency?.contract !== zeroAddress
-        ? (currency?.contract as UseBalanceToken)
-        : undefined,
-    watch: open,
-    formatUnits: currency?.decimals,
-  })
+  const paymentTokens = usePaymentTokens(
+    address !== undefined,
+    address as Address,
+    currency
+      ? {
+          symbol: currency.symbol as string,
+          address: currency?.address as Address,
+          name: currency?.name,
+          decimals: currency.decimals as number,
+          chainId: currency.chainId,
+        }
+      : chainCurrency,
+    totalPriceRaw - (feeOnTopRaw ?? 0n),
+    cartChain?.id,
+    false,
+    true
```

Comment:
Need to pass in the listing currency to the `usePaymentTokens` hook to fix that alwaysIncludeListingCurrency bug that we fixed in this PR: [https://github.com/reservoirprotocol/reservoir-kit/pull/472](https://github.com/reservoirprotocol/reservoir-kit/pull/472)

Reply:
Done, thanks.